src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopyColumnOrderHintCollection.cs

```diff
@@ -0,0 +1,84 @@
+            AssertWriteAccess();
+            if (string.IsNullOrEmpty(bulkCopyColumnOrderHint.Column) ||
+                bulkCopyColumnOrderHint.SortOrder == SortOrder.Unspecified)
+            {
+                throw SQL.BulkLoadInvalidOrderHint();
+            }
+            InnerList.Add(bulkCopyColumnOrderHint);
+            return bulkCopyColumnOrderHint;
+        }
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Add[@name="columnStringAndsortOrderSortOrder"]/*'/>
+        public SqlBulkCopyColumnOrderHint Add(string column, SortOrder sortOrder)
+        {
+            AssertWriteAccess();
+            return Add(new SqlBulkCopyColumnOrderHint(column, sortOrder));
+        }
+
+        private void AssertWriteAccess()
+        {
+            if (ReadOnly)
+            {
+                throw SQL.BulkLoadOrderHintInaccessible();
+            }
+        }
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Clear/*'/>
+        public new void Clear()
+        {
+            AssertWriteAccess();
+            base.Clear();
+        }
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Contains/*'/>
+        public bool Contains(SqlBulkCopyColumnOrderHint value) => InnerList.Contains(value);
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/CopyTo/*'/>
+        public void CopyTo(SqlBulkCopyColumnOrderHint[] array, int index) => InnerList.CopyTo(array, index);
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/IndexOf/*'/>
+        public int IndexOf(SqlBulkCopyColumnOrderHint value) => InnerList.IndexOf(value);
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Insert/*'/>
+        public void Insert(int index, SqlBulkCopyColumnOrderHint value)
+        {
+            AssertWriteAccess();
+            InnerList.Insert(index, value);
+        }
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Remove/*'/>
+        public void Remove(SqlBulkCopyColumnOrderHint value)
+        {
+            AssertWriteAccess();
+            InnerList.Remove(value);
+        }
```

Comment:
I did not see any tests for these functions. I know they are simple, but there is no guarantee for the future.

Reply:
They're covered in the functional tests, in SqlBulkCopyColumnOrderHintCollectionTest.cs.